catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/validation/impl/MatchOneValidatorTest.java

```diff
@@ -0,0 +1,140 @@
+
+    private static final AttributeImpl INVALID_ATTRIBUTE = new AttributeImpl(Core.TITLE, "robert");
+
+    private static final Set<String> VALID_ENUMERATIONS = ImmutableSet.of("bob", "rob", "bobby");
+
+    private static final Set<String> VALID_ENUMERATIONS_2 = ImmutableSet.of("ben");
+
+    private MatchOneValidator matchOneValidator;
+
+    private EnumerationValidator stubEmptyValidator;
+
+    private EnumerationValidator enumerationValidator;
+
+    private EnumerationValidator enumerationValidator2;
+
+    @Before
+    public void setUp() {
+        stubEmptyValidator = mock(EnumerationValidator.class);
+        when(stubEmptyValidator.validate(Matchers.any(AttributeImpl.class))).thenReturn(Optional.of(
+                new AttributeValidationReportImpl()));
+
+        enumerationValidator = new EnumerationValidator(VALID_ENUMERATIONS, false);
+        enumerationValidator2 = new EnumerationValidator(VALID_ENUMERATIONS_2, false);
+        SizeValidator sizeValidator = new SizeValidator(1, 3);
+        matchOneValidator = new MatchOneValidator(Arrays.asList(enumerationValidator,
+                sizeValidator));
+    }
+
+    @Test
+    public void testNullValidatorList() {
+        matchOneValidator = new MatchOneValidator(null);
+        Optional<AttributeValidationReport> attributeValidationReportOptional =
+                matchOneValidator.validate(VALID_ATTRIBUTE_1);
+        assertThat(attributeValidationReportOptional.isPresent(), is(false));
+    }
+
+    @Test
+    public void testNullAttribute() {
+        Optional<AttributeValidationReport> attributeValidationReportOptional =
+                matchOneValidator.validate(null);
+        assertThat(attributeValidationReportOptional.isPresent(), is(false));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testEmptyValidatorList() {
+        matchOneValidator = new MatchOneValidator(Collections.EMPTY_LIST);
+        Optional<AttributeValidationReport> attributeValidationReportOptional =
+                matchOneValidator.validate(VALID_ATTRIBUTE_1);
+        assertThat(attributeValidationReportOptional.isPresent(), is(false));
+    }
+
+    @Test
+    public void testEmptyValidationReport() {
+        matchOneValidator = new MatchOneValidator(Collections.singletonList(stubEmptyValidator));
+        Optional<AttributeValidationReport> attributeValidationReportOptional =
+                matchOneValidator.validate(VALID_ATTRIBUTE_1);
+        assertThat(attributeValidationReportOptional.isPresent(), is(false));
+    }
+
+    @Test
+    public void testValidationWithFailures() {
+        Optional<AttributeValidationReport> attributeValidationReportOptional =
+                matchOneValidator.validate(INVALID_ATTRIBUTE);
+        AttributeValidationReport attributeValidationReport =
+                attributeValidationReportOptional.get();
+        assertThat(attributeValidationReport.getAttributeValidationViolations()
```

Comment:
I think you could also do `assertThat(attributeValidationReport.getAttributeValidationViolations(), is(not(empty())))`.